api/src/adapters/aws/sesv2_adapter.py

```diff
@@ -22,22 +25,23 @@ class SESV2Response(BaseModel):
 
 class BaseSESV2Client(ABC, metaclass=ABCMeta):
     @abstractmethod
-    def list_suppressed_destinations(self) -> SESV2Response:
+    def list_suppressed_destinations(self, start_date: datetime | None = None) -> SESV2Response:
         pass
 
 
 class SESV2Client(BaseSESV2Client):
     def __init__(self) -> None:
-        self.client = boto3.client("sesv2")
+        self.client = get_sesv2_client()
 
     def list_suppressed_destinations(self, start_date: datetime | None = None) -> SESV2Response:
         request_params = {}
         if start_date:
             request_params["StartDate"] = start_date
 
         response = self.client.list_suppressed_destinations(**request_params)
+        response_object = SESV2Response.model_validate(response["SuppressedDestinationSummaries"])
```

Comment:
Because it always seems to cause issues when parsing other AWS/boto responses, what does a response look like if it's an error? Lets say we passed in nonsense to the API and it had an error response, what would we get back here?

While we probably still want an exception thrown, we probably don't want a key error. Maybe something like:
```suggestion
        if "SuppressedDestinationSummaries" in response:
            response_object = SESV2Response.model_validate(response["SuppressedDestinationSummaries"])
       else:
            # do some parsing on whatever it actually returned
            raise SomeError("...")
```